parsers/cxx/cxx_scope.c

```diff
@@ -227,11 +227,7 @@ void cxxScopePushTop(CXXToken * t)
 	cxxTokenChainAppend(g_pScope,t);
 	g_bScopeNameDirty = true;
 
-#ifdef CXX_DO_DEBUGGING
-	const char * szScopeName = cxxScopeGetFullName();
-
-	CXX_DEBUG_PRINT("Pushed scope: '%s'",szScopeName ? szScopeName : "");
-#endif
+	CXX_DEBUG_PRINT("Pushed scope: '%s'",cxxScopeGetFullName() ?: "");
```

Comment:
`?:` is GNU extension.

Reply:
Yes, it is a GNU extension. I like it but it should not be used in u-ctags.

Reply:
The double check requires only in built-time. So the double check is acceptable.
With the following code, we can avoid the double check but it is a bit harder to read.

```
#ifdef CXX_DO_DEBUGGING
	const char * szScopeName = cxxScopeGetFullName();
#endif
	CXX_DEBUG_PRINT("Pushed scope: '%s'",szScopeName ? szScopeName : "");
```